veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,185 @@
+
+## Repos
+
+- [https://github.com/kubevirt/kubevirt](https://github.com/kubevirt/kubevirt)
+
+## Design
+
+### Set TCP_REPAIR Socket Option
+
+During migration, active TCP sessions must transfer from one passt instance to another on a different host. The new instance cannot simply "hijack" the connection even if it knows all necessary details (IPs, ports, sequences). A privileged operation using the `TCP_REPAIR` socket option must be performed on both source and target sockets.
+
+Since passt runs in the unprivileged Linux context of virt-launcher, it lacks the necessary `CAP_NET_ADMIN` capability. Therefore, passt delegates this operation to a privileged helper running in virt-handler.
+
+#### The Role of passt-repair
+
+The helper responsible for handling `TCP_REPAIR` is called [passt-repair](https://passt.top/passt/tree/passt-repair.c), a C program included in the passt package.
+
+#### How passt-repair Works:
+- Executed as a **one-off process**, separately on migration source and migration target.
+- Communicates with passt over a named Linux Domain Socket.
+- Signals availability and waits for passt requests.
+- Upon receiving a request:
+  - Sets or unsets the `TCP_REPAIR` socket option.
+  - Notifies passt upon completion.
+
+#### Integration with virt-handler
+
+-To support this process, virt-handler will be modified to invoke `passt-repair` in a dedicated goroutine. This goroutine will
+run to completion or timeout, ensuring the migration process does not hang indefinitely.
+
+The invocation is conditioned on the VMI having an interface with `passt` network binding and an enabled feature gate:
+
+1. Before migration starts (source node) in [vmUpdateHelperMigrationSource](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-handler/vm.go#L2714) to set the TCP_REPAIR socket option.
+2. Before migration finalizes (target node) in [finalizeMigration](https://github.com/kubevirt/kubevirt/blob/release-1.5/pkg/virt-handler/vm.go#L2714) to set and then clear the TCP_REPAIR socket option.
+
+##### Failure Scenarios:
+passt-repair can fail or timeout if:
+- The Unix Domain Socket created by passt isn't found.
+- Communication with passt is delayed or unresponsive.
+
+Either scenario likely results in broken TCP connections for the guest VM, requiring reconnection.
+- virt-handler does not fail migration on passt-repair errors.
+- passt currently does not abort migration upon TCP_REPAIR failure.
+
+**Note**: Success of passt-repair doesn't guarantee seamless migration alone; sticky IPs from the network provider and CNI are essential.
+
+#### Add passt-repair binary to virt-handler image
+The `passt-repair` binary must be added to the virt-handler container image. The simplest solution is to install the passt RPM. A more secure and storage-efficient solution is extracting only the passt-repair binary from the RPM and installing it standalone.
+
+### Implement vhost-user support in passt sidecar
```

Comment:
Same, please provide a name here that reflects what this is aiming to solve and not how. Inside you should explain how this is accomplish.